### PR TITLE
Teams Adaptive Card Tabs

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes https://github.com/microsoft/botbuilder-dotnet/issues/5081

### CODE REVIEW

**Comment by mrivera-ms** [comment]
@chrimc62, could you please this?

**Comment by mrivera-ms** [comment]
@johnataylor to review this.

**Review by johnataylor — APPROVED** [review]

File: libraries/Microsoft.Bot.Schema/Teams/Generated/TabContext.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Schema.Teams
+{
+    using Newtonsoft.Json;
+    
+    /// <summary>
+    /// Current tab reqeust context, i.e., the current theme.
```
Comment (mdrichardson) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/TabContext.cs:9]:
```suggestion
    /// Current tab request context, i.e., the current theme.
```

File: libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCard.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace Microsoft.Bot.Schema.Teams
+{
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+    
+    /// <summary>
+    /// Envelope for cards for a Tab request.
+    /// </summary>
+    public partial class TabResponseCard
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCard.cs:12]: >partial  [](start = 11, length = 8)

Why is this partial?

File: libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCard.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Schema.Teams
+{
+    using System.Collections.Generic;
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCard.cs:6]: Doesn't look like this is needed.

File: libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCards.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Schema.Teams
+{
+    using System.Collections.Generic;
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCards.cs:6]: Not needed.

File: libraries/Microsoft.Bot.Schema/Teams/Generated/TaskModuleCardResponse.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            : base("continue")
+        {
+            CustomInit();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TaskModuleCardResponse"/> class.
+        /// </summary>
+        /// <param name="value">The JSON for the Adaptive cards to appear in the tab.</param>
+        public TaskModuleCardResponse(TabResponse value = default(TabResponse))
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/TaskModuleCardResponse.cs:26]: >(TabResponse) [](start = 65, length = 13)

This is redundant. You can only say "=default"

File: libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCard.cs (hunk 2 above)
Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCard.cs:12]: Many/most of the .schmea files are partial classes, and have a CustomInit which can be called during construction.  I think this is part of the ServiceClient pattern.

File: libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCards.cs (hunk 4 above)
Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Schema/Teams/Generated/TabResponseCards.cs:6]: used by the IList in this class

**Review by mrivera-ms — APPROVED** [review]
:shipit:

**Pull request merged**